soc/arm/nxp_imx/rt/soc_rt11xx.c

```diff
@@ -414,22 +414,15 @@ static ALWAYS_INLINE void clock_init(void)
 #endif
 #endif
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(enet1g), okay)
-	/*
-	 * 50 MHz clock for 10/100Mbit RMII PHY -
-	 * operate ENET1G just like ENET peripheral
-	 */
+	/* 125 MHz ENET1G clock */
 	rootCfg.mux = kCLOCK_ENET2_ClockRoot_MuxSysPll1Div2;
-	rootCfg.div = 10;
+	rootCfg.div = 4;
 	CLOCK_SetRootClock(kCLOCK_Root_Enet2, &rootCfg);
-#if CONFIG_ETH_MCUX_RMII_EXT_CLK
+	/* Set ENET1G TX_CLK to be driven by ENET2_CLK_ROOT and output on TX_CLK_IO pad */
+	IOMUXC_GPR->GPR5 = (IOMUXC_GPR_GPR5_ENET1G_RGMII_EN(0x01U) |
+		(IOMUXC_GPR->GPR5 & ~IOMUXC_GPR_GPR5_ENET1G_TX_CLK_SEL(0x01U)));
```

Comment:
since this is in SOC level, should we check the phy connection type property to determine the clocking ? maybe the enet1G is not always used with RGMII ?

Reply:
Updated, if it is not rgmii, the original code is called.